File: leetcode/easy/1_Two Sum.ts
function twoSum(nums: number[], target: number): number[] {
  const myHash: { [key: number]: number } = {};
  for (let i = 0; i < nums.length; i++) {
    myHash[target - nums[i]] = i;
  }
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] in myHash && i != myHash[nums[i]]) {
      return [i, myHash[nums[i]]];
    }
  }
  return [];
}

File: leetcode/easy/1_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numToIndex = {}
        for i,num in enumerate(nums):
            want = target - num
            if want in numToIndex:
                return [i,numToIndex[want]]
            numToIndex[num] = i
        return []
